runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpPanelLabel.java

```diff
@@ -55,6 +55,11 @@
 	 */
 	public String getActionKey(XpSnapshotSingle snapshot)
 	{
+		if (this == TIME_TO_LEVEL)
```

Comment:
I feel like changing `key`'s type to `Function<XpSnapshotSingle, String>` and creating an additional ctor for plain string arguments is a better approach to special-casing this. Then the change would be as simple as changing "TTL" to `(snap) -> snap.isHasGoal() ? "TTG" : "TTL"`.